application/actions/requests_test.go

```diff
@@ -1273,3 +1273,197 @@ func (as *ActionSuite) verifyRequestCreateInput(ctx context.Context, input api.R
+		},
+		{
+			name:       "good input",
+			user:       f.Users[0],
+			input:      goodRequest,
+			request:    f.Requests[2],
+			wantStatus: http.StatusOK,
+		},
+	}
+	for _, tt := range tests {
+		as.T().Run(tt.name, func(t *testing.T) {
+			req := as.JSON("/requests/" + tt.request.UUID.String())
+			req.Headers["Authorization"] = fmt.Sprintf("Bearer %s", tt.user.Nickname)
+			req.Headers["content-type"] = "application/json"
+			res := req.Put(&tt.input)
+
+			body := res.Body.String()
+			as.Equal(tt.wantStatus, res.Code, "incorrect status code returned, body: %s", body)
+
+			if tt.wantStatus == http.StatusOK {
+				wantData := []string{
+					`"created_by":{"id":"` + tt.user.UUID.String(),
+					`"organization":{"id":"` + f.Organization.UUID.String(),
+					`"description":"` + goodRequest.Description.String,
+				}
+				as.verifyResponseData(wantData, body)
+			}
+		})
+	}
+}
+
+func (as *ActionSuite) Test_convertUpdateRequestInput() {
+	userFixtures := test.CreateUserFixtures(as.DB, 1)
+	requests := test.CreateRequestFixtures(as.DB, 2, false, userFixtures.Users[0].ID)
+
+	creator := userFixtures.Users[0]
+
+	ctx := test.CtxWithUser(creator)
+
+	min := api.RequestUpdateInput{}
+
+	destination := api.Location{Description: "destination", Country: "XX"}
+	file := test.CreateFileFixture(as.DB)
+	origin := api.Location{Description: "origin", Country: "ZZ"}
+	size := api.RequestSize(models.RequestSizeMedium)
+	title := "request title"
+	visibility := api.RequestVisibility(models.RequestVisibilityAll)
+	full := api.RequestUpdateInput{
+		Description:  nulls.NewString("request description"),
+		Destination:  &destination,
+		Kilograms:    nulls.NewFloat64(1.0),
+		NeededBefore: nulls.NewTime(time.Now()),
+		Origin:       &origin,
+		PhotoID:      nulls.NewUUID(file.UUID),
+		Size:         &size,
+		Title:        &title,
+		Visibility:   &visibility,
+	}
+
+	tests := []struct {
+		name    string
+		input   api.RequestUpdateInput
+		request models.Request
+	}{
+		{
+			name:    "minimal",
+			input:   min,
+			request: requests[0],
+		},
+		{
+			name:    "full",
+			input:   full,
+			request: requests[1],
+		},
+	}
+	for _, tt := range tests {
+		as.T().Run(tt.name, func(t *testing.T) {
+			apiRequest, err := convertRequestUpdateInput(ctx, tt.input, tt.request.UUID.String())
+			as.NoError(err)
+
+			as.NoError(as.DB.Load(&tt.input))
+			as.verifyRequestUpdateInput(ctx, tt.input, tt.request, apiRequest)
+		})
+	}
+}
+
+func (as *ActionSuite) verifyRequestUpdateInput(ctx context.Context, input api.RequestUpdateInput, oldRequest, newRequest models.Request) {
+	as.Equal(models.CurrentUser(ctx).ID, newRequest.CreatedByID, "CreatedBy is not correct")
+
+	as.Equal(input.Description, newRequest.Description, "Description is not correct")
+
+	var destination models.Location
+	as.NoError(as.DB.Find(&destination, newRequest.DestinationID))
+	if msg := "Destination is not correct"; input.Destination != nil {
```

Comment:
It took me a minute to figure out why you had a mixed conditional. I'm not sure if it's worth it.  
I guess it gains you a line of code and allows using `msg :=` everywhere, instead of `msg :=` the first time and `msg =` after that.

It's interesting that the scope of `msg` must get down into the `else` block.

Reply:
OK, I wondered if it would be too strange